File: algorithms/cpp/389.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    char findTheDifference(string s, string t) {
        vector<int> cnt(26, 0);
        for (auto ch: s)
            cnt[ch-'a'] += 1;
        for (auto ch: t)
            cnt[ch-'a'] -= 1;
        char result;
        for (int i = 0; i < 26; i++)
            if (cnt[i] < 0)
                result = 'a'+i;
        return result;
    }
};

int main() {
    return 0;
}

File: algorithms/rust/389.rs
impl Solution {
    #[allow(dead_code)]
    pub fn find_the_difference(s: String, t: String) -> char {
        let mut s: Vec<char> = s.chars().collect();
        let mut t: Vec<char> = t.chars().collect();

        s.sort();
        t.sort();

        for (a, b) in s.iter().zip(t.iter()) {
            if a != b {
                println!("{:?}, {:?}", a, b);
                return *b;
            }
        }
        *t.last().unwrap()
    }
}

struct Solution;

fn main() {
    println!("Hello, world!");
}
